### PR TITLE
Add view ctors that don't take erased sentinel parameter

### PR DESCRIPTION
**Pull request by PointKernel**
This PR adds `static_map::device_view`/`static_map::device_mutable_view` constructors that don't take the erased sentinel parameter.

It's a dependency of https://github.com/rapidsai/cudf/pull/10983

### CODE REVIEW

File: include/cuco/static_map.cuh [hunk 1]
```diff
@@ -434,14 +434,26 @@ class static_map {
...
-                                         Key erased_key_sentinel) noexcept
+                                         sentinel::empty_key<Key> empty_key_sentinel,
+                                         sentinel::empty_value<Value> empty_value_sentinel) noexcept
       : slots_{slots},
         capacity_{capacity},
-        empty_key_sentinel_{empty_key_sentinel},
-        erased_key_sentinel_{erased_key_sentinel},
-        empty_value_sentinel_{empty_value_sentinel}
+        empty_key_sentinel_{empty_key_sentinel.value},
+        erased_key_sentinel_{empty_key_sentinel.value},
```
Comment (sleeepyjack) [inline include/cuco/static_map.cuh:436]: Ok, so we set the `erased_key_sentinel` to be the same as the `empty_key_sentinel`. This implies, that we must not call `map.erase(...)` or we break our probing scheme. In order to ensure this, we have a guard in the host API [here](https://github.com/PointKernel/cuCollections/blob/76dff03985ad224e068a0077eabeb6cbb26068a9/include/cuco/detail/static_map.inl#L161).
However, as far as I can see, there is no such guard for the device API.
Is this by design?

**Review by sleeepyjack — APPROVED** [review]

File: include/cuco/static_map.cuh (hunk 1 above)
Comment (jrhemstad) [reply include/cuco/static_map.cuh:436]: We don't really have a good way to guard it on the device API. The best would could probably do is an `assert`.

Comment (sleeepyjack) [reply include/cuco/static_map.cuh:436]: Agreed. We should at least add a disclaimer to the doxygen docs, which are btw missing for the device-sided `erase(…)` functions [here](https://github.com/NVIDIA/cuCollections/blob/76dff03985ad224e068a0077eabeb6cbb26068a9/include/cuco/static_map.cuh#L934) and [here](https://github.com/NVIDIA/cuCollections/blob/76dff03985ad224e068a0077eabeb6cbb26068a9/include/cuco/static_map.cuh#L940).

Comment (PointKernel) [reply include/cuco/static_map.cuh:436]: Docs added.

**Pull request merged**